lib/src/status.rs

```diff
@@ -200,6 +200,7 @@ pub(crate) fn get_status_require_booted(
 pub(crate) fn get_status(
     sysroot: &Storage,
     booted_deployment: Option<&ostree::Deployment>,
+    slot: Option<Slot>,
```

Comment:
I'm OK with this as is although...it feels like it'd be cleaner to have a method that filters the status after we gather it?

Something like this:

```
impl Host {
  fn filter_to_slot(&mut self, slot: Slot) {
     ...
}
```

It'd require less Option-juggling and also we wouldn't need to change all the callers.

Reply:
The other alternative is to pass the slot down to the rendering code instead.

Reply:
Force pushed the code into 1418f0b.  There wasn't much need in keeping the old commits around since all bar one line was changed.

Confirmed this still does work as expected, so please let me know if there's anything you feel could do with changing.